src/transformers/modeling_dpr.py

```diff
@@ -0,0 +1,537 @@
+)
+class DPRQuestionEncoder(DPRPretrainedQuestionEncoder):
+    r"""
+    Outputs: The DPR encoder only outputs the `pooler_output` that corresponds to the question representation:
+        **pooler_output**: ``torch.FloatTensor`` of shape ``(batch_size, hidden_size)``
+            Last layer hidden-state of the first token of the sequence (classification token)
+            further processed by a Linear layer. This output is to be used to embed questions for
+            nearest neighbors queries with context embeddings.
+
+    Examples::
+
+        tokenizer = DPRTokenizer.from_pretrained('dpr-base-uncased')
+        model = DPRQuestionEncoder.from_pretrained('dpr-ctx_encoder-base')
+        input_ids = torch.tensor(tokenizer.encode("Hello, is my dog cute ?")).unsqueeze(0)  # Batch size 1
+        embeddings = model(input_ids)  # the embeddings of the given question.
+
+    """
+
+    def __init__(self, config: DPRConfig):
+        super().__init__(config)
+        self.config = config
+        self.question_encoder = DPREncoder(config)
+        self.init_weights()
+
+    def forward(
+        self, input_ids: Tensor, token_type_ids: Optional[Tensor] = None, attention_mask: Optional[Tensor] = None
+    ) -> Tensor:
+        if attention_mask is None:
+            attention_mask = input_ids != self.config.pad_token_id
+            attention_mask = attention_mask.to(device=input_ids.device)
+        sequence_output, pooled_output, hidden_states = self.question_encoder(
+            input_ids, token_type_ids, attention_mask
+        )
+        return pooled_output
+
+
+DPRReaderOutput = collections.namedtuple(
+    "SpanPrediction", ["span_score", "relevance_score", "doc_id", "start_index", "end_index"]
+)
+
+
+@add_start_docstrings(
+    "The bare DPRReader transformer outputting span predictions.", DPR_START_DOCSTRING, DPR_READER_INPUTS_DOCSTRING,
+)
+class DPRReader(DPRPretrainedReader):
+    r"""
+    Outputs: `Tuple` comprising various elements depending on the configuration (config) and inputs:
+        **input_ids**: ``torch.FloatTensor`` of shape ``(n_passages, sequence_length)``
+            They correspond to the combined `input_ids` from `(question + context title + context content`).
+        **start_logits**: ``torch.FloatTensor`` of shape ``(n_passages, sequence_length)``
+            Logits of the start index of the span for each passage.
+        **end_logits**: ``torch.FloatTensor`` of shape ``(n_passages, sequence_length)``
+            Logits of the end index of the span for each passage.
+        **relevance_logits**: `torch.FloatTensor`` of shape ``(n_passages, )``
+            Outputs of the QA classifier of the DPRReader that corresponds to the scores of each passage
+            to answer the question, compared to all the other passages.
+
+
+    Examples::
+
+        tokenizer = DPRReader.from_pretrained('dpr-base-uncased')
+        model = DPRModel.from_pretrained('dpr-reader-base')
+        question_and_titles_ids = [
```

Comment:
nit => all examples could make use of the new tokenizer api : `tokenizer(...)`

Reply:
Here also you can add the import for model and tokenizer and also remove the `torch.tensor` in favor of a `return_tensors='pt'`

Reply:
Agree with that